pkg/prometheus/promcfg.go

```diff
@@ -1611,16 +1615,16 @@ func generateRunningFilter() yaml.MapSlice {
 	}
 }
 
-func getLimit(user *uint64, enforced *uint64) *uint64 {
+func getLimit(user *uint64, enforced *uint64, isGlobalLimitNotSupported bool) *uint64 {
 	if enforced == nil {
 		return user
 	}
 
-	if user == nil {
+	if isGlobalLimitNotSupported && user == nil {
 		return enforced
 	}
```

Comment:
My understanding is that when Prometheus supports global limits then we don't want to set scrape limit == enforced.
I find it also easier to have the version inline here.

```suggestion
	// Prometheus >= 2.45.0 supports global limits in which case the global limit gets applied for all targets that don't specify an explicit limit.
	if cg.version.GTE(semver.MustParse("2.45.0")) && ptr.Deref(user, 0) == 0 {
		return nil
	}
```